crates/ruma-federation-api/src/transactions/edu.rs

```diff
@@ -341,6 +342,43 @@ mod test {
         );
     }
 
+    #[test]
+    fn minimal_device_list_update_edu() {
+        let json = json!({
+                "content": {
+                    "device_id": "QBUAZIFURK",
+                    "stream_id": 6,
+                    "user_id": "@john:example.com"
+                },
+            "edu_type": "m.device_list_update"
+        });
+
+        let edu = serde_json::from_value::<Edu>(json).unwrap();
+        assert_matches!(
+            &edu,
+            Edu::DeviceListUpdate(DeviceListUpdateContent {
+                user_id,
+                device_id,
+                device_display_name,
+                stream_id,
+                prev_id,
+                deleted,
+                keys
+            }) if user_id == &user_id!("@john:example.com")
+                && device_id == &device_id!("QBUAZIFURK")
+                && device_display_name.is_none()
+                && stream_id == &uint!(6)
+                && prev_id.is_empty()
+                && deleted.is_none()
+                && keys.is_none()
+        );
+
+        assert_eq!(
+            serde_json::to_string(&edu).unwrap(),
+            r#"{"edu_type":"m.device_list_update","content":{"user_id":"@john:example.com","device_id":"QBUAZIFURK","stream_id":6}}"#
+        );
```

Comment:
Would be easier to read with `serde_json::to_value` & a `json!()` invocation.

Reply:
I've changed it in this test and the others in this file, for consistency.